packages/flutter/test/material/checkbox_test.dart

```diff
@@ -2331,3 +2360,31 @@ class _SelectedGrabMouseCursor extends MaterialStateMouseCursor {
   @override
   String get debugDescription => '_SelectedGrabMouseCursor()';
 }
+
+Widget buildAdaptiveCheckbox({
+  required TargetPlatform platform,
+  RoundedRectangleBorder shape = const RoundedRectangleBorder(),
+  BorderSide side = const BorderSide(),
+  bool value = true,
+  bool enabled = true,
+}) {
+  final Widget adaptiveCheckbox = Checkbox.adaptive(
+    value: value,
+    onChanged: enabled ? (_) {} : null,
+    shape: shape,
+    side: side,
+  );
+
+  return MaterialApp(
+    theme: ThemeData(platform: platform),
+    home: StatefulBuilder(
+      builder: (BuildContext context, StateSetter setState) {
+        return Material(
+          child: Center(
+            child: adaptiveCheckbox,
+          ),
+        );
+      },
+    ),
+  );
+}
```

Comment:
Some ideas for more things to test:

 * Any Material theming that will apply on iOS/macOS now? I mentioned visualDensity above, but maybe that's only controlled by the parameter?
 * The new parameters that used to be ignored.
 * As much as possible that is covered in the [CupertinoCheckBox test](https://github.com/flutter/flutter/blob/master/packages/flutter/test/cupertino/checkbox_test.dart).

Reply:
Points 1 and 2 look good. For point 3 `CupertinoCheckbox` doesn't have any painting tests I can find, only semantics/general behavior tests (can be focused by keyboard shortcuts, etc) that are already covered here.